guest/score/score.go

```diff
@@ -58,10 +58,10 @@ var _ func() uint64 = _score
 //
 //export score
 func _score() uint64 {
-	if score == nil {
-		// If we got here, someone imported the package, but forgot to set the
-		// filter. Panic with what's wrong.
-		panic("score imported, but score.SetPlugin not called")
+	if score == nil { // Then, the user didn't define one.
+		// This is likely caused by use of plugin.Set(p), where 'p' didn't
+		// implement ScorePlugin: return success and score zero.
+		return 0
```

Comment:
So, if you use the simple `plugin.Set(plugin)` we have to return something for score. Returning zero here (both status and score). There's no way to avoid this in simple mode because the host will call the function regardless. Only other way is complicated, but exposing a bitflag of functions that could be called.

I think that's too much, so if someone needs to assign score, really they should implement it. If they must not call score, they should not use `plugin.Set(p)` as the docs mention.